app/scripts/components/app/header.js

```diff
@@ -25,7 +26,14 @@ var Header = React.createClass({
     location: React.PropTypes.object,
     dispatch: React.PropTypes.func,
     api: React.PropTypes.object,
-    minimal: React.PropTypes.bool
+    minimal: React.PropTypes.bool,
+    apiVersion: React.PropTypes.object
+  },
+
+  componentWillMount: () => {
+    const { dispatch } = this.props;
+    dispatch(getApiVersion);
+    dispatch(validateApiVersion);
```

Comment:
* only do this dispatch if we're authenticated -`if this.props.api.authenticated`
* validateApiVersion can be done inside of the reducer
* could also look at `actions/index` mmt listCollections as an example